core/units.c

```diff
@@ -183,3 +186,49 @@ const struct units *get_units()
 {
 	return &prefs.units;
 }
+
+int parse_duration(const char *buffer, duration_t *time, bool dot_as_seperator)
+{
+	char *line = (char*)buffer;
+	size_t n = 0;
+	long values[3];
+	char separator;
+	while (n < sizeof(values)/sizeof(values[0])) {
+		values[n++] = strtol(line, &line, 10);
+		separator = *line;
+		line++;
+		/* special case of `mm.ss`
+		 * DivingLog 5.08 (and maybe other versions) appear to sometimes
+		 * store the dive time as 44.00 instead of 44:00 */
+		if (dot_as_seperator && n == 1 && separator == '.')
+			continue;
+		if (separator != ':')
+			break;
+	}
+	switch (n) {
+	case 1:
+		/* mm */
+		time->seconds = values[0] * 60;
+		time->ms = 0;
+		goto exit_out;
+	case 2:
+		/* mm:ss */
+		time->seconds = values[0] * 60 + values[1];
+		break;
+	case 3:
+		/* hh:mm:ss */
+		time->seconds = values[0] * 360 + values[1] * 60 + values[2];
+		break;
+	default:
+		time->seconds = 0;
+		time->ms = 0;
+		report_info("Strange sample time reading %s", buffer);
+		return 0;
+	}
+	long ms = 0;
+	if (separator == '.')
+		ms = strtol(line, &line, 10);
+	time->ms = ms;
```

Comment:
So the last separator was not a `.`, we skipped it. However, if the last separator was a `.` we parse milliseconds, but don't skip the next character.

This feels inconsistent to me.